### PR TITLE
Updates for https://github.com/cclib/cclib/pull/349

### PR DESCRIPTION
**Pull request by berquist**
I moved the Splitter to a more general place since it isn't Q-Chem only, handled the itertools problem, and fixed another parsing problem I encountered.

### CODE REVIEW

File: src/cclib/parser/utils.py [hunk 1]
```diff
@@ -143,6 +165,29 @@ def __init__(self):
             self.number[self.element[i]] = i
 
 
+class Splitter:
```
Comment (langner) [inline src/cclib/parser/utils.py]: Possibly rename to WidthSplitter?

File: src/cclib/parser/utils.py [hunk 2]
```diff
@@ -143,6 +165,29 @@ def __init__(self):
...
+        self.end_indices = list(accumulate(widths))
+
+    def split(self, line):
+        """Split the given line using the field widths passed in on class
+        initialization.
+        """
+        elements = [line[start:end].strip()
+                    for (start, end) in zip(self.start_indices, self.end_indices)]
+        for i in range(1, len(elements)):
+            if elements[-1] == '':
```
Comment (langner) [inline src/cclib/parser/utils.py]: I don't quite understand why there's a elements[-1] here. Perhaps add a test case to illustrate this?

Comment (berquist) [reply src/cclib/parser/utils.py]: Ok. The reason is to handle lines with empty regions, which come out like:

```python
['60', 'H', '10', 's', '0.14639', '0.00000', '0.00000', '-0.00000', '-0.00000', '0.00000']
['1', 'C', '1', 's', '-0.00000', '-0.00000', '0.00000', '', '', '']
```

and are then removed.

Comment (langner) [reply src/cclib/parser/utils.py]: I see. Perhaps make this optional, controlled by a named argument (can be True by default).

Also, is there a more concise form for this? Something like:
```
while len(element) and elements[-1] == '':
  elements.pop()
```

**Comment by langner** [comment]
I don't think Travis runs our doctests... could you add a proper unit test?

**Comment by langner** [comment]
Looks good!

**Pull request closed without merging**